packages/openid4vc/src/openid4vc-issuer/OpenId4VcIssuerService.ts

```diff
@@ -158,6 +164,15 @@ export class OpenId4VcIssuerService {
 
     const issuer = await this.getIssuerByIssuerId(agentContext, options.issuanceSession.issuerId)
 
+    const cNonce = getCNonceFromCredentialRequest(credentialRequest)
+    if (issuanceSession.cNonce !== cNonce) {
+      throw new CredoError('The cNonce in the credential request does not match the cNonce in the issuance session.')
+    }
+
+    if (!issuanceSession.cNonceExpiresAt || Date.now() > issuanceSession.cNonceExpiresAt.getTime()) {
```

Comment:
Might be good to separate the `!issuanceSession.cNonceExpiresAt` statement and give a more verbose error message which explains that the `cNonce` was not included, but is required.

Reply:
Hmm yeah should not really happen this case. They're always set together, mostly to make TS happy here, but i can split them up